lib/src/dbus_ws_client.dart

```diff
@@ -0,0 +1,256 @@
+part of 'dbus_client.dart';
+
+/// A client connection to a D-Bus over websocket server.
+class DBusWSClient extends DBusClient {
+  WebSocketChannel? _channel;
+  final _readJsonBuffer = DBusWSReadBuffer();
+  final Uri _uri;
+  DBusWSClient(this._uri) : super(DBusAddress("unix:path=/bus")) {}
+
+
+  /// Terminates all active connections. If a client remains unclosed, the Dart process may not terminate.
+  @override
+  Future<void> close() async {
+    if (_channel != null) {
+      await _channel!.sink.close();
+      _channel = null;
+      _connectCompleter = null;
+    }
+  }
+
+  /// Open a socket connection to the D-Bus server.
+  @override
+  Future<void> _openSocket() async {
+    // print("> openSocket");
+    _channel = WebSocketChannel.connect(_uri);
+    _channel?.stream.listen(_processDataJson);
+  }
+
+  /// Performs authentication with D-Bus server.
+  @override
+  Future<bool> _authenticate() async {
```

Comment:
somehow I can't find _authenticate() method in base classes (and this is supposed to '@override' it)